### PR TITLE
Add boilerplate components for transfer flow

### PR DESCRIPTION
**Pull request by delputnam**
#### Changes proposed in this Pull Request

We're getting ready to add a new transfer domain flow. This PR adds the boilerplate components for this flow and a feature flag to gate use of the new flow.

Note: This code is not complete, but only a starting point for additional features that will be added in subsequent PRs.

#### Testing instructions

Start the "Use a domain I own" flow with the `?flags=domains/new-transfer-flow` query string.

Inspect the code.

### CODE REVIEW

File: client/components/domains/connect-domain-step/page-definitions.js [hunk 1]
```diff
@@ -103,6 +107,44 @@ export const connectADomainOwnershipVerificationStepsDefinition = {
...
+		step: stepType.UNLOCK_DOMAIN,
+		name: __( 'Unlock domain' ),
+		component: TransferDomainStepUnlock,
+		next: stepSlug.TRANSFER_AUTH_CODE,
+		prev: stepSlug.TRANSFER_LOGIN,
+	},
+	[ stepSlug.TRANSFER_AUTH_CODE ]: {
+		mode: modeType.TRANSFER,
+		step: stepType.ENTER_AUTH_CODE,
+		name: __( 'Get domain authorization code' ),
```
Comment (a8ci18n) [inline client/components/domains/connect-domain-step/page-definitions.js:136]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'Get a domain authorization code.' )` *ES Score: 10*

File: client/components/domains/connect-domain-step/page-definitions.js [hunk 2]
```diff
@@ -103,6 +107,44 @@ export const connectADomainOwnershipVerificationStepsDefinition = {
...
+		component: TransferDomainStepAuthCode,
+		next: 'unused transfer domain step',
+		prev: stepSlug.TRANSFER_UNLOCK,
+	},
+	[ 'unused transfer domain step' ]: {
```
Comment (rafaelgallani) [inline client/components/domains/connect-domain-step/page-definitions.js:141]: I liked this solution for the breadcrumb 😄

**Review by rafaelgallani — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6637828

Hi @delputnam, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Unlock domain`
- `Finalize transfer`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6637828).

**Pull request merged**